packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -141,28 +141,78 @@ function createResourceEntry(
 }
 
 /**
- * Add a LCP event to the replay based on an LCP metric.
+ * Add a LCP event to the replay based on a LCP metric.
  */
 export function getLargestContentfulPaint(metric: {
   value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
   entries: PerformanceEntry[];
-}): ReplayPerformanceEntry<LargestContentfulPaintData> {
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'largest-contentful-paint');
+}
+
+/**
+ * Add a CLS event to the replay based on a CLS metric.
+ */
+export function getCumulativeLayoutShift(metric: {
+  value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
+  entries: PerformanceEntry[];
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'cumulative-layout-shift');
+}
+
+/**
+ * Add a FID event to the replay based on a FID metric.
+ */
+export function getFirstInputDelay(metric: {
+  value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
+  entries: PerformanceEntry[];
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'first-input-delay');
+}
+
+/**
+ * Add an INP event to the replay based on an INP metric.
+ */
+export function getInteractionToNextPaint(metric: {
+  value: number;
+  rating: 'good' | 'needs-improvement' | 'poor';
+  entries: PerformanceEntry[];
+}): ReplayPerformanceEntry<WebVitalData> {
+  return getWebVital(metric, 'interaction-to-next-paint');
+}
+
+/**
+ * Add an web vital event to the replay based on the web vital metric.
+ */
+export function getWebVital(
+  metric: {
+    value: number;
+    rating: 'good' | 'needs-improvement' | 'poor';
+    entries: PerformanceEntry[];
+  },
+  name: string,
+): ReplayPerformanceEntry<WebVitalData> {
   const entries = metric.entries;
   const lastEntry = entries[entries.length - 1] as (PerformanceEntry & { element?: Element }) | undefined;
   const element = lastEntry ? lastEntry.element : undefined;
 
   const value = metric.value;
+  const rating = metric.rating;
 
   const end = getAbsoluteTime(value);
 
-  const data: ReplayPerformanceEntry<LargestContentfulPaintData> = {
-    type: 'largest-contentful-paint',
-    name: 'largest-contentful-paint',
+  const data: ReplayPerformanceEntry<WebVitalData> = {
+    type: 'web-vital',
+    name,
     start: end,
     end,
     data: {
       value,
       size: value,
+      rating,
       nodeId: element ? record.mirror.getId(element) : undefined,
```

Comment:
Curious, do we end up getting node ids for every web vital?

Reply:
we do!

Reply:
we do for all of them except cls